hydra-plutus/src/Hydra/Contract/HeadState.hs

```diff
@@ -24,17 +24,20 @@ data State
   = Initial
       { contestationPeriod :: ContestationPeriod
       , parties :: [Party]
+      , initialHeadId :: CurrencySymbol
       }
   | Open
       { contestationPeriod :: ContestationPeriod
       , parties :: [Party]
       , utxoHash :: Hash
+      , openHeadId :: CurrencySymbol
       }
   | Closed
       { parties :: [Party]
       , snapshotNumber :: SnapshotNumber
       , utxoHash :: Hash
       , contestationDeadline :: POSIXTime
+      , closedHeadId :: CurrencySymbol
```

Comment:
I wanted for us to be able to see in code from which constructor exactly the value comes since it gets passed to different validator checks and could be useful when debugging.

Reply:
Ok but look at this piece of code where we have `CloseObservation`, `ClosedState` and `OpenState` all having `headId` field:

```
  let CloseObservation{threadOutput, headId, snapshotNumber} = observation
  guard (headId == headId)
  let ClosedThreadOutput{closedContestationDeadline} = threadOutput
  let event =
        OnCloseTx
          { snapshotNumber
          , contestationDeadline = posixToUTCTime closedContestationDeadline
          }
  let st' =
        ClosedState
          { closedThreadOutput = threadOutput
          , headId = headId
          , closedHeadTokenScript = openHeadTokenScript
          }
  pure (event, st')
 where
  OpenState
    { headId
    , openHeadTokenScript
    } = st

```
I can introduce qualified names but introducing different field names felt natural to me.